kernel/src/scan/mod.rs

```diff
@@ -690,23 +690,46 @@ pub(crate) mod test_utils {
         Box::new(ArrowEngineData::new(batch))
     }
 
-    // simple add
-    pub(crate) fn add_batch_simple() -> Box<ArrowEngineData> {
+    // simple sidecar
+    pub(crate) fn sidecar_batch_with_given_paths(
+        paths: Vec<&str>,
+        output_schema: SchemaRef,
+    ) -> Box<ArrowEngineData> {
+        let handler = SyncJsonHandler {};
+
+        let json_strings: StringArray = paths
+        .iter()
+        .map(|path| {
+            format!(
+                r#"{{"sidecar":{{"path":"{path}","sizeInBytes":9268,"modificationTime":1714496113961,"tags":{{"tag_foo":"tag_bar"}}}}}}"#,
+            )
+        })
+        .collect::<Vec<_>>()
```

Comment:
Could avoid the type annotation
```suggestion
        .collect_vec()
  ```

Reply:
itertools +1